service/lib/agama/storage/proposal.rb

```diff
@@ -89,6 +89,8 @@ def storage_json
           source_json || {
             legacyAutoyastStorage: JSON.parse(strategy.settings.to_json, symbolize_names: true)
           }
+        else
+          {}
```

Comment:
On a call we have clarified that the `null` from D-Bus is desired, the Rust client should be fixed to handle it instead